api/v1alpha1/ndb_api_helpers.go

```diff
@@ -159,43 +160,151 @@ func GetNoneTimeMachineSLA(ctx context.Context, ndbclient *ndbclient.NDBClient)
 	return sla, fmt.Errorf("NONE TimeMachine not found")
 }
 
+// +kubebuilder:object:generate:=false
+type ProfileResolver interface {
+	resolve(ctx context.Context, allProfiles []ProfileResponse, filter func(allProfiles ProfileResponse) bool) (profile ProfileResponse, err error)
+}
+
+func (providedProfile *Profile) resolve(ctx context.Context, allProfiles []ProfileResponse, filter func(allProfiles ProfileResponse) bool) (profile ProfileResponse, err error) {
+	log := ctrllog.FromContext(ctx)
+	log.Info("resolving...", "provided profile", providedProfile)
```

Comment:
We can be explicit the fileName.functionName in the log to be consistent with the other logs..
log.Info("Entered ndb_api_helpers.resolve")